src/containers/booksForm.js

```diff
@@ -1,11 +1,80 @@
-/* eslint-disable react/prefer-stateless-function */
+/* eslint-disable class-methods-use-this */
 /* eslint-disable no-plusplus */
+
 import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { createBook } from '../actions/index';
+
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSelect = this.handleSelect.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    event.preventDefault();
+    this.setState({
+      title: event.target.value,
+    });
+  }
+
+  handleSelect(event) {
+    event.preventDefault();
+    this.setState({
+      category: event.target.value,
+    });
+  }
+
+  validate(title, category) {
+    const formError = document.getElementById('formError');
+    if (title === '') {
+      formError.innerHTML = "Title can't be empty";
+      setTimeout(() => {
+        formError.innerHTML = '';
+      }, 3000);
+      return false;
+    }
+    if (category === '') {
+      formError.innerHTML = "Category can't be empty";
+      setTimeout(() => {
+        formError.innerHTML = '';
+      }, 3000);
+      return false;
+    }
+    return true;
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const { title, category } = this.state;
+    if (!this.validate(title, category)) return;
+
+    const { createBook } = this.props;
+    const newBook = {
+      id: Math.floor(Math.random() * 10000),
+      title,
+      category,
+    };
+
+    createBook(newBook);
+
+    this.setState({
+      title: '',
+      category: '',
+    });
+  }
 
-export default class BooksForm extends React.Component {
   render() {
     const categoryOptions = [];
     const categoryNames = [
+      '',
```

Comment:
This should be `'All'`, right?
```suggestion
      'All',
```

Reply:
thanks but this is for the next milestone4 :)